app/models/package_manager/maven/maven_central.rb

```diff
@@ -20,4 +20,20 @@ def self.one_version(raw_project, version_string)
   def self.missing_version_remover
     PackageManager::Base::MissingVersionRemover
   end
+
+  # maven-metadata.xml for Maven Central does not appear to be guaranteed to contain all relevant versions for a package
+  # So instead, if needed, we will retrieve the versions from the raw HTML index page
+  def self.versions(raw_project, name)
+    if raw_project && raw_project[:versions]
+      raw_project[:versions]
+    else
+      retrieve_versions(versions_from_html(name), name)
```

Comment:
What happens if the repository is temporarily unavailable?

Reply:
If the repository is temporarily unavailable then I think this will just not retrieve the versions; which was already the case with the previous more generic method this was using seen here: https://github.com/librariesio/libraries.io/blob/main/app/models/package_manager/maven.rb#L159-L167

Reply:
Ok, as long as we are not marking everything as removed or something when there is a temporary server issue.

Reply:
I will investigate this a bit, because this new methodology will cause the same behavior in a temporary server issue as it would previously. But agreed that if it would mark everything as removed in that case, we probably should fix that too.